src/compiler/scanner.ts

```diff
@@ -286,7 +286,7 @@ namespace ts {
 
     const tokenStrings = makeReverseMap(textToToken);
 
-    export function tokenToString(t: SyntaxKind): string {
+    export function tokenToString(t: SyntaxKind): string | undefined {

```

Comment:
If you pass a SyntaxKind of a Node. And even if you only pass SyntaxKind of Tokens, you get undefined for example for StringLiteral, NumericLiteral, etc.